clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -157,6 +157,15 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
 
   mlir::Value VisitCastExpr(CastExpr *e);
 
+  mlir::Value VisitArraySubscriptExpr(ArraySubscriptExpr *e) {
+    if (e->getBase()->getType()->isVectorType()) {
+      assert(!cir::MissingFeatures::scalableVectors() &&
```

Comment:
```suggestion
      assert(!cir::MissingFeatures::scalableVectors());
```
The second part of this assert isn't necessary.

This does still need an errorNYI for vector types and a return without falling through to the call below. Vector types in general are handled in the incubator and not covered by the missing features assert here.